app/controllers/geocoder_controller.rb

```diff
@@ -228,39 +224,20 @@ def normalize_params
     params.permit(:query, :lat, :lon, :latlon_digits, :zoom, :minlat, :minlon, :maxlat, :maxlon)
   end
 
-  def nsew_to_decdeg(captures)
-    begin
-      Float(captures[0])
-      lat = captures[2].casecmp("s").zero? ? -captures[0].to_f : captures[0].to_f
-      lon = captures[5].casecmp("w").zero? ? -captures[3].to_f : captures[3].to_f
-    rescue StandardError
-      lat = captures[0].casecmp("s").zero? ? -captures[1].to_f : captures[1].to_f
-      lon = captures[3].casecmp("w").zero? ? -captures[4].to_f : captures[4].to_f
-    end
-    { :lat => lat, :lon => lon }
-  end
+  def to_decdeg(captures)
+    ns = captures.fetch("ns").casecmp("s").zero? ? -1 : 1
+    nsd = BigDecimal(captures.fetch("nsd", "0"))
+    nsm = BigDecimal(captures.fetch("nsm", "0"))
+    nss = BigDecimal(captures.fetch("nss", "0"))
 
-  def ddm_to_decdeg(captures)
-    begin
-      Float(captures[0])
-      lat = captures[3].casecmp("s").zero? ? -(captures[0].to_f + (captures[1].to_f / 60)) : captures[0].to_f + (captures[1].to_f / 60)
-      lon = captures[7].casecmp("w").zero? ? -(captures[4].to_f + (captures[5].to_f / 60)) : captures[4].to_f + (captures[5].to_f / 60)
-    rescue StandardError
-      lat = captures[0].casecmp("s").zero? ? -(captures[1].to_f + (captures[2].to_f / 60)) : captures[1].to_f + (captures[2].to_f / 60)
-      lon = captures[4].casecmp("w").zero? ? -(captures[5].to_f + (captures[6].to_f / 60)) : captures[5].to_f + (captures[6].to_f / 60)
-    end
-    { :lat => lat, :lon => lon }
-  end
+    ew = captures.fetch("ew").casecmp("w").zero? ? -1 : 1
+    ewd = BigDecimal(captures.fetch("ewd", "0"))
+    ewm = BigDecimal(captures.fetch("ewm", "0"))
+    ews = BigDecimal(captures.fetch("ews", "0"))
 
-  def dms_to_decdeg(captures)
-    begin
-      Float(captures[0])
-      lat = captures[4].casecmp("s").zero? ? -(captures[0].to_f + ((captures[1].to_f + (captures[2].to_f / 60)) / 60)) : captures[0].to_f + ((captures[1].to_f + (captures[2].to_f / 60)) / 60)
-      lon = captures[9].casecmp("w").zero? ? -(captures[5].to_f + ((captures[6].to_f + (captures[7].to_f / 60)) / 60)) : captures[5].to_f + ((captures[6].to_f + (captures[7].to_f / 60)) / 60)
-    rescue StandardError
-      lat = captures[0].casecmp("s").zero? ? -(captures[1].to_f + ((captures[2].to_f + (captures[3].to_f / 60)) / 60)) : captures[1].to_f + ((captures[2].to_f + (captures[3].to_f / 60)) / 60)
-      lon = captures[5].casecmp("w").zero? ? -(captures[6].to_f + ((captures[7].to_f + (captures[8].to_f / 60)) / 60)) : captures[6].to_f + ((captures[7].to_f + (captures[8].to_f / 60)) / 60)
-    end
-    { :lat => lat, :lon => lon }
+    lat = ns * (nsd + (nsm / 60) + (nss / 3600))
+    lon = ew * (ewd + (ewm / 60) + (ews / 3600))
+
+    { :lat => lat.to_s("F"), :lon => lon.to_s("F") }
```

Comment:
Need to `.truncate()` to some reasonable precision before `.to_s` or else the length blows up:

![image](https://github.com/openstreetmap/openstreetmap-website/assets/4158490/38852968-1c8f-4d02-bda7-0fd871a02eef)

Reply:
Fixed... 
![image](https://github.com/openstreetmap/openstreetmap-website/assets/147741/e367ee1f-133a-4372-b709-8e7405d81d94)